backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecHandler.scala

```diff
@@ -57,6 +60,45 @@ import scala.collection.mutable.ArrayBuffer
 
 class SparkPlanExecHandler extends SparkPlanExecApi {
 
+  /**
+   * Transform GetArrayItem to Substrait.
+   *
+   * arrCol[index] => IF(index < 0, null, ElementAt(arrCol, index + 1))
+   */
+  override def genGetArrayItemExpressionNode(
+      substraitExprName: String,
+      functionMap: java.util.HashMap[String, java.lang.Long],
+      leftNode: ExpressionNode,
+      rightNode: ExpressionNode,
+      original: GetArrayItem): ExpressionNode = {
+    // ignore origin substraitExprName
+    val functionName = ConverterUtils.makeFuncName(
+      ExpressionMappings.expressionsMap.get(classOf[ElementAt]).get,
+      Seq(original.dataType),
+      FunctionConfig.OPT)
+    val exprNodes = Lists.newArrayList(
+      leftNode.asInstanceOf[ExpressionNode],
+      rightNode.asInstanceOf[ExpressionNode])
+    val resultNode = ExpressionBuilder.makeScalarFunction(
+      ExpressionBuilder.newScalarFunction(functionMap, functionName),
+      exprNodes,
+      ConverterUtils.getTypeNode(original.dataType, original.nullable))
+    val nullNode = ExpressionBuilder.makeLiteral(null, original.dataType, false)
+    val lessThanFuncId = ExpressionBuilder.newScalarFunction(
+      functionMap,
+      ConverterUtils.makeFuncName(
+        ExpressionNames.LESS_THAN,
+        Seq(original.right.dataType, IntegerType),
+        FunctionConfig.OPT))
+    // right node already add 1
+    val literalNode = ExpressionBuilder.makeLiteral(1.toInt, IntegerType, false)
+    val lessThanFuncNode = ExpressionBuilder.makeScalarFunction(
+      lessThanFuncId,
+      Lists.newArrayList(rightNode, literalNode),
+      ConverterUtils.getTypeNode(BooleanType, true))
+    new IfThenNode(Lists.newArrayList(lessThanFuncNode), Lists.newArrayList(nullNode), resultNode)
+  }
+
```

Comment:
Why not implement Spark `element_at` at velox side? Current solution seems like a workaround at transformer layer and it hides the difference between Spark `get_array_item` and Presto `element_at`. Furthermore, we may have to do such a workaround for Map `element_at` in future.

Reply:
Actually, the `element_at` function in Spark is consistent with the one in Velox. It is the`get_array_item` function in Spark that has some semantic differences with the `element_at` function in Spark. 
Here’s an example to illustrate the difference between `get_array_item` and `element_at` in Spark.
GetArrayItem UT:
https://github.com/apache/spark/blob/master/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ComplexTypeSuite.scala#L63
element_at UT:
https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala#L2130

Reply:
So do we still need to do implementation of `get_array_item` at velox side? I see you don't change the function name in transformer.

Reply:
AFAIK, it is not nessesary to support `get_array_item` in velox now. It seems that current implement has already solve the difference between `get_array_tiem` and `element_at`.

Reply:
You resolve the difference here but still need to map `get_array_item` to `element_at` for offloading to velox, as changes in [2760](https://github.com/oap-project/gluten/pull/2760). The current status is fallback. You can confirm and I will reopen the PR if need.

Reply:
After apply this pr, it'll not fallback

Reply:
> So do we still need to do implementation of `get_array_item` at velox side? I see you don't change the function name in transformer.

function name changed in `genGetArrayItemExpressionNode`